client/app/components/dashboards/DashboardGrid.jsx

```diff
@@ -33,6 +33,53 @@ const WidgetType = PropTypes.shape({
 const SINGLE = "single-column";
 const MULTI = "multi-column";
 
+const DashboardWidget = React.memo(
+  function DashboardWidget({
+    widget,
+    dashboard,
+    onLoadWidget,
+    onRefreshWidget,
+    onRemoveWidget,
+    onParameterMappingsChange,
+    canEdit,
+    isPublic,
+    isLoading,
+    filters,
+  }) {
+    const { type } = widget;
+    const onLoad = () => onLoadWidget(widget);
+    const onRefresh = () => onRefreshWidget(widget);
+    const onDelete = () => onRemoveWidget(widget.id);
+
+    if (type === WidgetTypeEnum.VISUALIZATION) {
+      return (
+        <VisualizationWidget
+          widget={widget}
+          dashboard={dashboard}
+          filters={filters}
+          canEdit={canEdit}
+          isPublic={isPublic}
+          isLoading={isLoading}
+          onLoad={onLoad}
+          onRefresh={onRefresh}
+          onDelete={onDelete}
+          onParameterMappingsChange={onParameterMappingsChange}
+        />
+      );
+    }
+    if (type === WidgetTypeEnum.TEXTBOX) {
+      return <TextboxWidget widget={widget} canEdit={canEdit} isPublic={isPublic} onDelete={onDelete} />;
+    }
+    return <RestrictedWidget widget={widget} />;
+  },
+  (prevProps, nextProps) =>
+    prevProps.widget === nextProps.widget &&
+    prevProps.canEdit === nextProps.canEdit &&
+    prevProps.isPublic === nextProps.isPublic &&
+    prevProps.isLoading === nextProps.isLoading &&
+    prevProps.filters === nextProps.filters
+);
```

Comment:
I didn't directly removed the `dashboard` prop from the widget (didn't seem that easy), but I selected what props will trigger the rerender instead.

Reply:
Is `filters` prop immutable? 
Why move `isLoading` out of `widget`?

Reply:
> Is filters prop immutable?

From my understanding, yes, it's defined in a state every time `loadDashboard` and the dashboard doesn't mutate it. I quickly opened a dashboard with filters and looks like it's working.

I think I'll push a PR with some Cypress tests for filters. It should be a good thing , since we don't test Filters all the time.

> Why move isLoading out of widget?

This leads to the one that most concerns me and the reason to that: the widget isn't an immutable prop, hence why I moved `isLoading` out of widget. We can have a mutable prop, but we need to make sure that all render dependent props trigger a change when they should. And that's mostly what concerns me about testing this PR. From my tests and from existing Cypress coverage, the `isLoading` looks like the only one.